scripts/ingest_model.py

```
@@ -302,61 +304,118 @@ def run_archive(model: Model, checks: Checklist) -> None:
     if run_cmd(*uv_run_args("--with pytest pytest tests/site/test_fig_payloads.py -q")):
         checks.ok("Multi-model payloads regenerated + shape tests passed")
     else:
         checks.fail("Payload shape tests failed after regeneration")
 
 
-def map_yaml_paths(paths: Sequence[str]) -> list[str]:
-    """Map model YAML paths (as in a PR diff) to generated enum names.
+def read_head_yaml(path: str) -> dict[str, object] | None:
+    """Read one model YAML from HEAD, returning None when it is newly added."""
+    git_command = ["git", "show", f"HEAD:{path}"]
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)